feat: union sealed class

Miqqy0069 opened the pull request:
- added actions file for before and after
- added generic async file for Loading, Error, Data
- modified overview page, implementing the Loading Action

(working app -> error vid)
https://user-images.githubusercontent.com/75653480/227851451-29b6b713-15a2-4f07-8cd4-270b3abbda81.mp4

(analyze) 
![image_360](https://user-images.githubusercontent.com/75653480/227846183-625bb841-679d-4009-9bb3-0d0cc0f9fdaf.png)

(dart format)
![image_720](https://user-images.githubusercontent.com/75653480/227846187-b9502515-a9f4-4670-b0c4-4b24c45bc914.png)

Miqqy0069 committed:
feat: union sealed class
- added actions file for before and after
- added generic async file for Loading, Error, Data
- modified overview page, implementing the Loading Action

Miqqy0069 committed:
feat: union sealed class
- removed hard coded string on error message

shawn-ffufm commented on lib/features/pokemon_overview/pokemon_overview_page.dart:
```diff
@@ -9,24 +10,34 @@ class PokemonOverviewPage extends StatelessWidget {
-          mainAxisSpacing: 10,
+      body: pokemons.when(
+        loading: () => const Center(child: CircularProgressIndicator()),
+        error: (String? errorMessage) => Center(
```
errorMessage is enough, no need to include its type 🙂

Miqqy0069 replied:
resolved

shawn-ffufm commented on lib/features/pokemon_overview/pokemon_overview_page.dart:
```diff
@@ -9,24 +10,34 @@ class PokemonOverviewPage extends StatelessWidget {
+        loading: () => const Center(child: CircularProgressIndicator()),
+        error: (String? errorMessage) => Center(
+          child: Text(
+            errorMessage!,
```
instead of enforcing, use null coalescing here then pass an empty string instead so it's safer 🙂

Miqqy0069 replied:
resolved

shawn-ffufm commented on lib/state/action/actions.dart:
```diff
@@ -0,0 +1,14 @@
+import 'package:async_redux/async_redux.dart';
+import 'package:pokedex_async_redux/state/app_state.dart';
+
+abstract class LoadingAction extends ReduxAction<AppState> {
```
you forgot to add a concise documentation 😝

remember, this is mandtoy right? 😉

Miqqy0069 replied:
resolved

shawn-ffufm commented on lib/state/action/pokemon_actions.dart:
```diff
@@ -1,29 +1,19 @@
-}
-
-/// Getting the pokemon Details from pokemon api
-class GetPokemonDetailsAction extends ReduxAction<AppState> {
```
why removed this one? you only need to update `GetPokemonsAction` right? 😱

Miqqy0069 replied:
i was just testing it po xd , i'll just restore it

Miqqy0069 replied:
resolved

shawn-ffufm reviewed (changes requested):
see pr feedback 🙂

Miqqy0069 committed:
feat: union sealed class
(pr feedback)
- removed datatype for error message on overview page
- replaced enforce to null coalescing to pass a empty string
- added documentation to LoadingAction class
- restored getpokemondetailsaction

shawn-ffufm reviewed (approved):
lgtm
